### ISSUE TITLE
Feature request: dual layers

### ISSUE THREAD

**Issue by morgan3d** [deepnight/led#103]
I'd like to have a layer where the tiles are placed on _edges_ instead of cells, for placing doors, bars, and thin walls that are still grid aligned.

This is a little tricky because there are 2x as many edges as cells (if you ignore an off-by-one of the border), so it isn't just an offset layer for mouse coords and rendering. However, it could be modeled internally as two layers, where one is offset vertically by half a cell and the other is offset horizontally, but drawing paints to both simultaneously so that it feels to the artist as if it is a single edge layer.

**Comment by deepnight** [comment]
You could have a Tile layer with a smaller grid than the Tileset acctual grid (eg. a layer of 8px grid with a tileset of 16px). Would it work for you?

**Comment by morgan3d** [comment]
Good point!

A tile layer with a smaller grid works if the tools can be set for that layer to mask out every other tile, like masking the black tiles on a checkerboard.

That way the regular tools and pattern rules will only write to the legal locations on that layer (edges) and not to the centers and corners that are being ignored. This sounds like the easiest implementation strategy; an "edge" layer is really just a higher resolution later that refuses to write to cells where (x + y) & 1 != 0.

[Note that this is **also** an alternative and pretty elegant way of implementing hex grids, which map 1:1 to white checkerboard tiles.]

**Comment by deepnight** [comment]
I actually might have a much simpler solution for your case: I already support internal offsetX/Y for each layers (because of level resizing issues with various grid sizes). Basically, just having 8px offset on a 16px grid should do the job, right?

**Comment by morgan3d** [comment]
Yes, if there are two offset layers [internally] and I can draw to them simultaneously using the tools, that's perfect!

In the Tiled editor I tried approximating this with two separate offset layers and it was too difficult for artists to use effectively. They had to switch active layer for vertical vs. horizontal details and kept getting confused and drawing on the wrong one, and fill commands & patterns didn't work properly of course.

**Comment by deepnight** [comment]
No, in my suggestion you have 1 layer with the normal 16px grid, and one with with the grid offset-ed by 8px.

With the future changes implied by #32 (freely positioned tiles), it will possible for me to add a **Snap to grid borders** mode, giving you what you're looking for :)
